### PR TITLE
Upgrade to API 32.0

### LINKED ISSUE #499: As a developer, recreate the lookup filter on Address__c.Household_Account__c once the build works with it, so users won't pick incorrect accounts for an address.
having this lookup filter causes our build to fail to deploy the package.  Jason believes it is a Salesforce bug.  Until we can resolve this, I removed the filter to unblock our work.

The filter looks like: 
Household Account: _SYSTEM: AccountType equals Value 'HH Account'
filter type: Enabled
error: Address objects can only be used with Household Accounts

we are waiting for API version 30 (spring '14) to see if SF core's redesign of named filters solves this issue.

### PR DESCRIPTION
**Pull request by jlantz**
This branch now passes all tests with a few adaptations to changes in 30.0, 31.0, and 32.0.

It sets the <apiVersion> element in all meta.xml files to 31.0 to ensure consistency and includes changes to the updateMetaXml ant target to update all meta.xml files to api version specified in cumulusci.properties (cumulus.project.apiVersion).

This branch also includes the configuration of lookup filters on Contact.npsp__Current_Address__c and npsp__Address__c.npsp__Household_Account__c

I think we want to do more testing of this before merging.  I'm creating the pull request so we can use it to track discussion of what other testing we want to do and the results of those tests.

# Warning
* This release changes all metadata from the package to run on API 32.0.  The change has undergone extensive testing, but custom code which extends the packaged metadata may behave differently.  Please test your custom functionality and third party add on products after upgrade and post any issues in the Power of Us Hub.

# Info
* The Contact.npsp__Current_Address__c field now restricts to only addresses related to the Contact's Household Account.
* The npsp__Address__c.npsp__Household_Account__c field now restricts to only Accounts which are Household Accounts.

# Issues
Fixes #499

### CODE REVIEW

**Comment by njjc** [comment]
@davidhabib ready for review and merging. Feel free to ping me to talk about anything. I'd suggest adding a ?w=1 at the end of the diff URL to ignore whitespace changes.

File: src/classes/CONV_Account_Conversion_CTRL.cls [hunk 1]
```diff
@@ -96,7 +96,7 @@ public with sharing class CONV_Account_Conversion_CTRL {
         for (String s : targetTokenMap.keyset()){
             if (s.endsWith('__c')){
                 List<String> nameSegs = s.split('__');
-                if (nameSegs.size() < 3 && Limits.getFieldsDescribes() < Limits.getLimitFieldsDescribes()){
```
Comment (njjc) [inline src/classes/CONV_Account_Conversion_CTRL.cls]: I'm not sure where this came from or if this is a regression. @kbromer

**Comment by njjc** [comment]
@davidhabib this is ready when you are; the latest build was a whitespace change so it shouldn't be an issue. Thanks!

File: src/classes/STG_PanelRelAuto_CTRL.cls [hunk 2]
```diff
@@ -133,19 +133,14 @@ public with sharing class STG_PanelRelAuto_CTRL extends STG_Panel {
...
-	                        Schema.DescribeFieldResult F = fieldTokenMap.get(s).getDescribe();             
-	                        List <Schema.sObjectType> sobjref = new List<Schema.sObjectType>(); 
-	                        sobjref = F.getReferenceTo();
-	                        for (Schema.sObjectType objT : sobjref) {
-	                            if (objT == Contact.sObjectType) {
-	                                listSOACfield.add(new SelectOption(F.getName(), F.getLabel()));
-	                            }                   
-	                        }               
-	                    }           
+	                    Schema.DescribeFieldResult F = fieldTokenMap.get(s).getDescribe();
```
Comment (davidhabib) [inline src/classes/STG_PanelRelAuto_CTRL.cls]: @njjc why did the describe check get removed?  i'm not saying we need it, but curious why it got yanked.

Comment (njjc) [inline src/classes/STG_PanelRelAuto_CTRL.cls]: An API version got rid of describe limits.

**Pull request merged**